src/migration/service/migrator.ts

```diff
@@ -31,6 +35,30 @@ export async function getSqlMigrationNames(migrationPath: string): Promise<strin
   return Array.from(migrationSet);
 }
 
+/**
+ * Glob the migration directory and retrieve all the migration entries (names)
+ * that needs to be run.
+ *
+ * @param {string} migrationPath
+ * @param {string} extension
+ * @returns {Promise<string[]>}
+ */
+export async function getJavaScriptMigrationNames(migrationPath: string, extension: string): Promise<string[]> {
+  const files = await glob(migrationPath);
+  const migrationSet = new Set<string>();
+  const pattern: RegExp = extension === FileExtensionTypes.JS ? FILE_PATTERN_JS : FILE_PATTERN_TS;
```

Comment:
Isn't `RegExp` automatically inferred?

```suggestion
  const pattern = extension === FileExtensionTypes.JS ? FILE_PATTERN_JS : FILE_PATTERN_TS;
```